WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java

```diff
@@ -264,13 +264,17 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
             renderPost(position, (ReaderPostViewHolder) holder);
         } else if (holder instanceof ReaderXPostViewHolder) {
             renderXPost(position, (ReaderXPostViewHolder) holder);
-        } else if (holder instanceof BlogInfoViewHolder) {
-            BlogInfoViewHolder blogHolder = (BlogInfoViewHolder) holder;
-            blogHolder.mBlogInfoView.setOnBlogInfoLoadedListener(mBlogInfoLoadedListener);
-            blogHolder.mBlogInfoView.loadBlogInfo(mCurrentBlogId, mCurrentFeedId);
-        } else if (holder instanceof TagInfoViewHolder) {
-            TagInfoViewHolder tagHolder = (TagInfoViewHolder) holder;
-            tagHolder.mTagInfoView.setCurrentTag(mCurrentTag);
+        } else if (holder instanceof SiteHeaderViewHolder) {
+            SiteHeaderViewHolder blogHolder = (SiteHeaderViewHolder) holder;
+            blogHolder.mSiteHeaderView.setOnBlogInfoLoadedListener(mBlogInfoLoadedListener);
+            if (isDiscover()) {
+                blogHolder.mSiteHeaderView.loadBlogInfo(ReaderConstants.DISCOVER_SITE_ID, 0);
```

Comment:
If the value were anything other than `0` I would definitely agree, but I think it's fairly common to assume `0` means "empty" for numeric variables. I'm fine to revisit this in a separate PR if you feel otherwise, though.